### ISSUE TITLE
[post frontend-backend split] Move types/functions from common to frontend/backend

### ISSUE THREAD

**Issue by ed255** [privacy-scaling-explorations/halo2#266]
After the frontend-backend split a new package `halo2_common` was added that contains functions and types used by both frontend and backend.  These functions/types are now `pub` but are meant for internal usage only of the halo2 repository.

As future work we should try to move more stuff to either frontend/backend, creating mirror lightweight types as necessary; or splitting functionality between frontend and backend.  After types/functions are moved to frontend/backend, they should recover the `pub(crate)` status.

Blocked by https://github.com/privacy-scaling-explorations/halo2/pull/254

**Comment by duguorong009** [comment]
I worked on task - move `halo2_common::arithmetic` to `halo2_backend`.  https://github.com/privacy-scaling-explorations/halo2/pull/281

**Comment by duguorong009** [comment]
I work on task - move `plonk::Challenge*` & `plonk::transcript` from `common` to `backend`. #287

**Comment by ed255** [comment]
@duguorong009 will work on the remaining points of this issue.

**Comment by duguorong009** [comment]
@ed255 
I've just merged the #298 .
After this, there is no `halo2_common` crate(the post-split tech debt) in the repo.
Hence, I think we can close this issue as completed. 

What do you think?

**Comment by ed255** [comment]
> @ed255 I've just merged the #298 . After this, there is no `halo2_common` crate(the post-split tech debt) in the repo. Hence, I think we can close this issue as completed.
> 
> What do you think?

I agree!  This issue has now been completed, I'll just close it; thanks for the reminder!
